OloEngine/src/OloEngine/Renderer/MeshOptimization.cpp

```
@@ -0,0 +1,528 @@
+        auto lodMesh = Ref<MeshSource>::Create(MoveTemp(lodVertices), MoveTemp(lodIndices));
+
+        // Copy material table from source
+        for (const auto& [index, handle] : meshSource.GetMaterials())
+        {
+            lodMesh->SetMaterial(index, handle);
+        }
+
+        const auto& srcSubmeshes = meshSource.GetSubmeshes();
+        if (srcSubmeshes.Num() == 1)
+        {
+            Submesh submesh = srcSubmeshes[0];
```

Clang (14.0.6)

[warning] variable 'submesh' is not initialized

(cppcoreguidelines-init-variables)